### PR TITLE
[`macro_metavar_expr_concat`] Dogfooding

### PR DESCRIPTION
**Pull request by c410-f3r**
cc #124225

Starts inner usage to test the robustness of the implementation.

### CODE REVIEW

File: library/std/src/sys/pal/unix/weak.rs [hunk 1]
```diff
@@ -189,10 +186,7 @@ pub(crate) macro raw_syscall {
             // (not paths).
             use libc::*;
 
-            syscall(
-                concat_idents!(SYS_, $name),
-                $($arg_name),*
-            ) as $ret
+            syscall(${concat(SYS_, $name)}, $($arg_name),*) as $ret
```
Comment (Noratrieb) [inline library/std/src/sys/pal/unix/weak.rs:189]:
```suggestion
            syscall(libc::${concat(SYS_, $name)}, $($arg_name),*) as $ret
```
same here

**Comment by c410-f3r** [comment]
Done

File: library/std/src/sys/pal/unix/weak.rs [hunk 2]
```diff
@@ -185,14 +178,7 @@ pub(crate) macro syscall {
...
         unsafe fn $name($($arg_name:$t),*) -> $ret {
-            // This looks like a hack, but concat_idents only accepts idents
-            // (not paths).
-            use libc::*;
-
-            syscall(
-                concat_idents!(SYS_, $name),
-                $($arg_name),*
-            ) as $ret
+            libc::syscall(libc::${concat(SYS_, $name)}, $($arg_name),*) as $ret
```
Comment (workingjubilee) [inline library/std/src/sys/pal/unix/weak.rs:181]: Out of curiosity (you do not have to change to this), would this work?
```rust
libc::syscall(${concat(libc::SYS_, $name)}, $($arg_name),*) as $ret
```

File: library/std/src/sys/pal/unix/weak.rs [hunk 3]
```diff
@@ -168,14 +168,7 @@ pub(crate) macro syscall {
             if let Some(fun) = $name.get() {
                 fun($($arg_name),*)
             } else {
-                // This looks like a hack, but concat_idents only accepts idents
-                // (not paths).
-                use libc::*;
```
Comment (workingjubilee) [inline library/std/src/sys/pal/unix/weak.rs:173]: no this absolutely was a brutal hack.

**Review by workingjubilee — APPROVED** [review]
Neat.

File: library/std/src/sys/pal/unix/weak.rs (hunk 3 above)
Comment (Noratrieb) [reply library/std/src/sys/pal/unix/weak.rs:173]: a beautiful example of why it eas terrible

**Pull request merged**